src/frame.c

```diff
@@ -542,7 +543,8 @@ void frame_process(frame_t *st, size_t length)
             }
             else
             {
-                input_pdu_push(st->input, st->buffer + offset, cnt, prog, hdr.stream_id);
+                input_pdu_push(st->input, st->buffer + offset, cnt, prog, hdr.stream_id, seq);
+                seq = (seq + 1) & 0x3f;
```

Comment:
The sequence number also needs to be incremented in the case where the CRC is invalid. This line won't be reached due to the `continue` on line 523.

Reply:
Fair point. Looking over this code, I feel like there are similar problems with `st->pdu_idx[prog][hdr.stream_id]` as well, so some refactoring might be warranted.

Reply:
Yes, you're right. We should reset `st->pdu_idx[prog][hdr.stream_id]` back to zero in the `j == 0 && hdr.pfirst` case, regardless of whether the CRC is valid.